src/Algebra/Graph/Acyclic/AdjacencyMap.hs

```diff
@@ -0,0 +1,405 @@
+-- @
+transitiveClosure :: Ord a => AdjacencyMap a -> AdjacencyMap a
+transitiveClosure (AAM x) = AAM (AM.transitiveClosure x)
+
+-- | Compute the /condensation/ of a graph, where each vertex
+-- corresponds to a /strongly-connected component/ of the original
+-- graph. Note that component graphs are non-empty, and are therefore
+-- of type "Algebra.Graph.NonEmpty.AdjacencyMap".
+--
+-- @
+-- scc 'empty'      == 'empty'
+-- scc ('vertex' x) == 'vertex' (NonEmpty.'NonEmpty.vertex' x)
+-- @
+scc :: (Ord a) => AM.AdjacencyMap a -> AdjacencyMap (NonEmpty.AdjacencyMap a)
+scc = AAM . AM.scc
+
+-- | Compute the /topological sort/ of a graph.
+-- 
+-- @
+-- topSort (1)             == [1]
+-- topSort (1 * 2 + 3 * 1) == [3,1,2]
+-- @
+topSort :: (Ord a) => AdjacencyMap a -> [a]
+topSort (AAM am) = Typed.topSort (Typed.fromAdjacencyMap am)
+
+-- | Compute the /Cartesian product/ of graphs.
+-- Complexity: /O(s1 * s2)/ time, memory and size, where /s1/ and /s2/
+-- are the sizes of the given graphs.
+--
+-- @
+-- edgeList   (box (1 * 2) (3 * 4)) == [((1,3),(1,4))
+--                                     ,((1,3),(2,3))
+--                                     ,((1,4),(2,4))
+--                                     ,((2,3),(2,4))]
+-- edgeList   (box (1 + 2) (3 + 4)) == []
+-- vertexList (box (1 + 2) (3 + 4)) == [(1,3),(1,4),(2,3),(2,4)]
+-- @
+box :: (Ord a, Ord b) => AdjacencyMap a -> AdjacencyMap b -> AdjacencyMap (a, b)
+box (AAM x) (AAM y) = AAM $ AM.box x y
+
+-- | Remove a vertex from a given acyclic graph.
+-- Complexity: /O(n*log(n))/ time.
+--
+-- @
+-- removeVertex x ('vertex' x)       == 'empty'
+-- removeVertex 1 ('vertex' 2)       == 'vertex' 2
+-- removeVertex 1 (1 * 2)          == 'vertex' 2
+-- removeVertex x . removeVertex x == removeVertex x
+-- @
+removeVertex :: Ord a => a -> AdjacencyMap a -> AdjacencyMap a
+removeVertex x = AAM . AM.removeVertex x . aam
+
+-- | Remove an edge from a given acyclic graph.
+-- Complexity: /O(log(n))/ time.
+--
+-- @
+-- removeEdge 1 2 (1 * 2)          == (1 + 2)
+-- removeEdge x y . removeEdge x y == removeEdge x y
+-- removeEdge x y . 'removeVertex' x == 'removeVertex' x
+-- removeEdge 1 2 (1 * 2 + 3 * 4)  == 1 + 2 + 3 * 4
+-- @
+removeEdge :: Ord a => a -> a -> AdjacencyMap a -> AdjacencyMap a
+removeEdge x y = AAM . AM.removeEdge x y . aam
+
+-- | This is a signature for a __Strict Partial Order__.
+-- A strict partial order is a binary relation __/R/__ that has three
+-- axioms, namely, irreflexive, transitive and asymmetric.
+--
+--   > a 'R' a == False               (Irreflexive)
+--   > a 'R' b and b 'R' c => a 'R' c (Transitive)
+-- Some examples of a Strict Partial Order are 
+-- __\<__ and __\>__.
+type PartialOrder a = a -> a -> Bool
+
+-- A helper function.
+-- This is not efficient but can be used till an efficient version
+-- of induce is included in the main library.
+induceE :: (Ord a) => ((a, a) -> Bool) -> AdjacencyMap a -> AdjacencyMap a
+induceE p a = rmER p eL a
```

Comment:
Hopefully, there will be an `induceE` in the `Graph.AdjacencyMap` module for me to use.